tests/runtime/test_deployment.py

```diff
@@ -29,9 +29,45 @@ async def test_known_attributes_autocomplete(self):
         assert "id" in dir(deployment)
         assert "foo" not in dir(deployment)
 
-    async def test_attribute_override_via_env_var(self, monkeypatch):
-        monkeypatch.setenv(name="PREFECT__RUNTIME__DEPLOYMENT__NEW_KEY", value="foobar")
-        assert deployment.new_key == "foobar"
+    @pytest.mark.parametrize(
+        "env_name, env_value, expected_value",
+        [
+            # new user defined env var
+            ("new_key", "foobar", "foobar"),
+            # KNOWN FIELDS
+            # id is of type str
+            ("id", "fake-id", "fake-id"),
+            # flow_run_id is of type str
+            ("flow_run_id", "fake-flow-run-id", "fake-flow-run-id"),
+        ],
+    )
+    async def test_attribute_override_via_env_var(
+        self, monkeypatch, env_name, env_value, expected_value
+    ):
+        monkeypatch.setenv(
+            name=f"PREFECT__RUNTIME__DEPLOYMENT__{env_name.upper()}", value=env_value
+        )
+        deployment_attr = getattr(deployment, env_name)
+        # check the type of the flow_run attribute
+        assert isinstance(deployment_attr, type(expected_value))
+        # check the flow_run attribute value is expected_value
+        assert deployment_attr == expected_value
+
+    @pytest.mark.parametrize(
+        "env_name",
+        [
+            # complex types (list and dict) not allowed to be mocked using environment variables
+            "parameters",
```

Comment:
This tests only `dict`, correct? If so, we would ideally have one for `list`.

Reply:
Yeah, the problem is that it only can be tested with existing attributes. In the case of deployments attributes are:
- id (Optional[str])
- flow_run_id (Optional[str])
- parameters (Dict)
- name (Optional[Dict])
- version (Optional[Dict])

As any of the attributes are of type List, it can be tested. It can be mocked if you think it would improve the readability, but the code that raises the exception is already tested when the type is Dict.

Reply:
👍

Reply:
@jcozar87 Looking through the tests a bit more, yes I do think we should mock types that don't exist, e.g.
```python 
           ...  # other parameters
           ("pretend_number", "123", 123),
           ("pretend_bool", "True", True),
        ],
    )
    async def test_attribute_override_via_env_var(
        self, monkeypatch, attribute_name, attribute_value, expected_value
    ):
        flow_run.FIELDS["pretend_number"] = int
        flow_run.FIELDS["pretend_bool"] = bool
```

Reply:
Hi @jcozar87, I am confused by the addition of the new test. Using the code above, it won't raise and you can ensure that a user is able to cast to a bool, int, float, etc., which would be ideal to ensure that works in case new runtime attribute types are added in future. I don't believe any new tests need to be added, just new parameters for each of the 3 test files. After that, we should be good to go!

Reply:
Hi @jcozar87, if you want to add a new test type to each file that is fine with me. As long as we test that all of the basic types can be correctly cast (bool, int, float, etc.) so that we can be sure we can provide that functionality to the user 🙂 Whether you want to add them in as a new test or new parameters is fine either way with me!